test/e2e/firewall.go

```diff
@@ -0,0 +1,410 @@
+
+	for _, allowed := range alloweds {
+		for _, port := range allowed.Ports {
+			resultProtocolPorts = append(resultProtocolPorts, strings.ToLower(allowed.IPProtocol+"/"+port))
+		}
+	}
+	for _, sp := range servicePorts {
+		expectedProtocolPorts = append(expectedProtocolPorts, strings.ToLower(string(sp.Protocol)+"/"+strconv.Itoa(int(sp.Port))))
+	}
+	return sameStringElements(resultProtocolPorts, expectedProtocolPorts)
+}
+
+func verifySourceRanges(resultSourceRanges, expectedSourceRanges []string) error {
+	if expectedSourceRanges == nil {
+		expectedSourceRanges = append(expectedSourceRanges, "0.0.0.0/0")
+	}
+	return sameStringElements(resultSourceRanges, expectedSourceRanges)
+}
+
+func sameStringElements(result, expected []string) error {
+	ex := make(map[string]bool)
+	for _, s := range expected {
+		_, ok := ex[s]
+		if ok {
+			return fmt.Errorf("duplicated element: %v", s)
+		}
+		ex[s] = true
+	}
+	for _, s := range result {
+		_, ok := ex[s]
+		if !ok {
+			return fmt.Errorf("unexpected element: %v", s)
+		}
+		delete(ex, s)
+	}
+	if len(ex) != 0 {
+		return fmt.Errorf("missing elements: %v", ex)
+	}
+	return nil
+}
+
+func createNetexecDaemonSet(c clientset.Interface, dsName, ns string, labels map[string]string, httpPort, udpPort int32, hostNetwork bool) error {
+	_, err := c.Extensions().DaemonSets(ns).Create(&extensions.DaemonSet{
+		ObjectMeta: api.ObjectMeta{
+			Name:   dsName,
+			Labels: labels,
+		},
+		Spec: extensions.DaemonSetSpec{
+			Template: api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: api.PodSpec{
+					NodeSelector: nil,
+					Containers: []api.Container{
+						{
+							Name:  dsName,
+							Image: framework.NetexecImageName,
+							Command: []string{
+								"/netexec",
+								fmt.Sprintf("--http-port=%d", httpPort),
+								fmt.Sprintf("--udp-port=%d", udpPort),
+							},
+							Ports: []api.ContainerPort{
+								{
+									Name:          "http",
+									ContainerPort: httpPort,
+								},
+								{
+									Name:          "udp",
+									ContainerPort: udpPort,
+								},
+							},
+						},
+					},
+					SecurityContext: &api.PodSecurityContext{
+						HostNetwork: hostNetwork,
+					},
+				},
+			},
+		},
+	})
+	return err
+}
+
+func deleteDaemonSet(f *framework.Framework, dsName, ns string, labels map[string]string) {
+	framework.Logf("Check that reaper kills all daemon pods for %s", dsName)
+	dsReaper, err := kubectl.ReaperFor(extensions.Kind("DaemonSet"), f.ClientSet)
+	Expect(err).NotTo(HaveOccurred())
+	err = dsReaper.Stop(ns, dsName, 0, nil)
+	Expect(err).NotTo(HaveOccurred())
+	err = wait.Poll(dsRetryPeriod, dsRetryTimeout, checkRunningOnNoNodes(f, labels))
+	Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to be reaped")
+}
+
+func getNonMaterNodeNames(c clientset.Interface) (sets.String, error) {
+	masterName := framework.TestContext.CloudConfig.MasterName
+	nodeList, err := c.Core().Nodes().List(api.ListOptions{})
```

Comment:
I don't think you want to get all nodes, see how the services test handles this (limits nodes in large clusters, because that will take hours to complete).